app/src/main/kotlin/dev/patrickgold/florisboard/lib/compose/ScrollableModifiers.kt

```diff
@@ -179,3 +186,86 @@ fun Modifier.florisScrollbar(
+    val alpha by animateFloatAsState(
+        targetValue = targetAlpha,
+        animationSpec = tween(durationMillis = duration, easing = ScrollbarAnimationEasing),
+    )
+    val scrollbarColor = color.takeOrElse { MaterialTheme.colors.onSurface.copy(alpha = 0.28f) }
+
+    LaunchedEffect(Unit) {
+        delay(1850)
+        isInitial = false
+    }
+
+    val orientation = state.layoutInfo.orientation
+    val visibleItemsInfo = state.layoutInfo.visibleItemsInfo
+    val visibleItems = if (visibleItemsInfo.isNotEmpty()) remember { visibleItemsInfo.size } else 0
+    val last = visibleItemsInfo.lastOrNull()
+    val stacks = if (last != null && orientation == Orientation.Vertical) {
+        remember { last.column }
+    } else if (last != null && orientation == Orientation.Horizontal) {
+        remember { last.row }
+    } else {
+        0
+    }
+    drawWithContent {
+        drawContent()
+        val firstVisibleElementIndex = state.layoutInfo.visibleItemsInfo.firstOrNull()?.index
+        val needDrawScrollbar = state.isScrollInProgress || isInitial || alpha > 0f
+        if (needDrawScrollbar && firstVisibleElementIndex != null) {
+            val scrollbarWidth: Float
+            val scrollbarHeight: Float
+            val scrollbarOffsetX: Float
+            val scrollbarOffsetY: Float
+            val first = state.layoutInfo.visibleItemsInfo.first()
+
+            if (orientation == Orientation.Vertical) {
+                val elementHeight = this.size.height / state.layoutInfo.totalItemsCount
+                scrollbarWidth = size.toPx()
+                scrollbarOffsetX = this.size.width - scrollbarWidth
+                scrollbarOffsetY = (firstVisibleElementIndex - stacks*percentOffset(first, orientation)) * elementHeight
+                scrollbarHeight = min(visibleItems * elementHeight, this.size.height - scrollbarOffsetY)
+            } else {
+                val elementWidth = this.size.width / state.layoutInfo.totalItemsCount
+                scrollbarHeight = size.toPx()
+                scrollbarOffsetX = (firstVisibleElementIndex - stacks*percentOffset(first, orientation)) * elementWidth
+                scrollbarOffsetY = this.size.height - scrollbarHeight
+                scrollbarWidth = min(visibleItems * elementWidth, this.size.height - scrollbarOffsetX)
+            }
+
+            drawRect(
+                color = scrollbarColor,
+                topLeft = Offset(scrollbarOffsetX, scrollbarOffsetY),
+                size = Size(scrollbarWidth, scrollbarHeight),
+                alpha = alpha,
+            )
+        }
+    }
+}
+
+/**
+ * Item's offset on main axis as a percentage of size
+ */
+internal fun percentOffset (
+    item: LazyListItemInfo,
+): Float {
+    return item.offset.toFloat() / item.size
+}
+
+internal fun percentOffset (
+    item: LazyGridItemInfo,
+    orientation: Orientation
+): Float {
+    val offset = if (orientation==Orientation.Horizontal) item.offset.x else item.offset.y
```

Comment:
I just saw that this could be changed to make the code more readable.
```suggestion
    val offset = if (orientation==Orientation.Horizontal) {
        item.offset.x else item.offset.y
    }
```